ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -46,6 +51,14 @@ const PermitSimulation: React.FC<{
     return undefined;
   }, [exchangeRate, value]);
 
+  const { tokenValue, tokenValueMaxPrecision } = useMemo(() => {
+    const valueBN = new BigNumber(value / Math.pow(10, tokenDecimals));
+    return {
+      tokenValue: formatAmount('en-US', valueBN),
+      tokenValueMaxPrecision: formatAmountMaxPrecision('en-US', valueBN),
+    };
+  }, [tokenDecimals, value]);
```

Comment:
Note there were some issues with this code. We are addressing the issues here:
- https://github.com/MetaMask/metamask-extension/issues/25751
- https://github.com/MetaMask/metamask-extension/issues/25755